iroh-gossip/src/net.rs

```diff
@@ -229,10 +229,15 @@ impl Gossip {
         }
     }
 
-    /// Handle an incoming [`quinn::Connection`].
+    /// Handle an incoming [`Connection`].
     ///
     /// Make sure to check the ALPN protocol yourself before passing the connection.
-    pub async fn handle_connection(&self, conn: quinn::Connection) -> anyhow::Result<()> {
+    ///
+    /// [`Connection`]: iroh_net::magic_endpoint::Connection
+    pub async fn handle_connection(
+        &self,
+        conn: iroh_net::magic_endpoint::Connection,
```

Comment:
we already do a bunch of `iroh_net` imports in these files. Why not do the same and keep the docs and params simply referring the import? The new state is a tad too verbose for my liking. What do you think?

Reply:
I went with a direct replacement as I was not editing the local code: if `quinn::SomeThing` was used I replaced it without imports as well.  It is true the new full name is a lot longer.  It will probably also change to `iroh_net::endpoint` sometime soon as well which is at least a bit shorter.

I will attempt some cleanup.

Reply:
290e447 is more aggressive with imports, hope this is better